### ISSUE TITLE
Ensure indentation_linter supports double-indented functions

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1754]
Slipped my mind during review of `indentation_linter()` that the Tidyverse guide now supports double-indented function signatures:

https://style.tidyverse.org/functions.html#long-lines-1

```r
foo <- function(
    a = 1,
    b = 2) {
  # body
}
```

I don't remember whether there are tests of this behavior, but we should support it by default if we don't already.

**Comment by AshesITR** [comment]
Do we also want to allow the following?

```r
function(
  a = 1
  b = 2
) {
  # body
}
```

**Comment by MichaelChirico** [comment]
personally I like that style but it's definitely not allowed by default (this was my original suggestion but decided against in style guide, can find the issue if you'd like)

does indentation linter allow that currently? I'm more ambiguous as to whether it should lint from indentation_linter... maybe it'd be covered by a function_signature_linter (I know there are a few other linters' that would fit there)

**Comment by AshesITR** [comment]
Indentation linter currently allows your preferred style (which is also mine).
